src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/NewAzureRmVpnClientIpsecParametersCommand.cs

```diff
@@ -0,0 +1,141 @@
+        [ValidateRange(1024, int.MaxValue)]
+        public int SADataSize { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IPSec encryption algorithm (IKE Phase 1)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IpsecEncryption.GCMAES256,
+            MNM.IpsecEncryption.GCMAES128,
+            MNM.IpsecEncryption.AES256,
+            MNM.IpsecEncryption.AES128,
+            IgnoreCase = false)]
+        public string IpsecEncryption { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IPSec integrity algorithm (IKE Phase 1)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IpsecIntegrity.GCMAES256,
+            MNM.IpsecIntegrity.GCMAES128,
+            MNM.IpsecIntegrity.SHA256,
+            IgnoreCase = false)]
+        public string IpsecIntegrity { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IKE encryption algorithm (IKE Phase 2)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IkeEncryption.GCMAES256,
+            MNM.IkeEncryption.GCMAES128,
+            MNM.IkeEncryption.AES256,
+            MNM.IkeEncryption.AES128,
+            IgnoreCase = false)]
+        public string IkeEncryption { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient IKE integrity algorithm (IKE Phase 2)")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.IkeIntegrity.SHA384,
+            MNM.IkeIntegrity.SHA256,
+            IgnoreCase = false)]
+        public string IkeIntegrity { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient DH Groups used in IKE Phase 1 for initial SA")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.DhGroup.DHGroup24,
+            MNM.DhGroup.ECP384,
+            MNM.DhGroup.ECP256,
+            MNM.DhGroup.DHGroup14,
+            MNM.DhGroup.DHGroup2,
+            IgnoreCase = false)]
+        public string DhGroup { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Vpnclient PFS Groups used in IKE Phase 2 for new child SA")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+            MNM.PfsGroup.PFS24,
+            MNM.PfsGroup.PFSMM,
+            MNM.PfsGroup.ECP384,
+            MNM.PfsGroup.ECP256,
+            MNM.PfsGroup.PFS14,
+            MNM.PfsGroup.PFS2,
+            MNM.PfsGroup.None,
+            IgnoreCase = false)]
+        public string PfsGroup { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (ShouldProcess(Properties.Resources.CreatingResourceMessage + Properties.Resources.VirtualNetworkGatewayVpnClientIpsecPolicy))
```

Comment:
@Nilambari it doesn't appear that this cmdlet doesn't make any calls to the server, it's just creating an in-memory object. If that is the case, then you don't need `ShouldProcess` and you can remove `SupportsShouldProcess` from the cmdlet attribute